lnrpc/routerrpc/router_server.go

```diff
@@ -347,30 +347,13 @@ func (s *Server) SendPaymentV2(req *SendPaymentRequest,
 		return err
 	}
 
-	// Send the payment.
-	err = s.cfg.Router.SendPaymentAsync(payment, paySession, shardTracker)
-	if err == nil {
-		// If the payment was sent successfully, we can start tracking
-		// the events.
-		return s.trackPayment(
-			sub, payHash, stream, req.NoInflightUpdates,
-		)
-	}
-
-	// Otherwise, transform user errors to grpc code.
-	if errors.Is(err, channeldb.ErrPaymentInFlight) ||
-		errors.Is(err, channeldb.ErrAlreadyPaid) {
-
-		log.Debugf("SendPayment async result for payment %x: %v",
-			payment.Identifier(), err)
+	// Send the payment asynchronously.
+	s.cfg.Router.SendPaymentAsync(payment, paySession, shardTracker)
 
-		return status.Error(codes.AlreadyExists, err.Error())
-	}
-
-	log.Errorf("SendPayment async error for payment %x: %v",
-		payment.Identifier(), err)
-
-	return err
+	// Track the payment and return.
+	return s.trackPayment(
+		sub, payHash, stream, req.NoInflightUpdates,
+	)
 }
```

Comment:
Should we keep the payment info log in case the payment fails? Maybe add it to `trackPayment`?
```
log.Errorf("SendPayment async error for payment %x: %v",
		payment.Identifier(), err)
```

Reply:
good q. updated the `trackPayment` to print the logs, also think he `context.Canceled` error can be safely ignored.